lib/kafka/protocol/message.rb

```diff
@@ -57,11 +57,12 @@ def decompress
         data = codec.decompress(value)
         message_set_decoder = Decoder.from_string(data)
         message_set = MessageSet.decode(message_set_decoder)
+        base_offset = offset - message_set.size + 1
```

Comment:
for example offset is 42 but size is 3, we want to start at 40

Reply:
This is just weird – what about the message that precedes the container message? Wouldn't that have offset 41?

Reply:
offset for the message precedes the container message is `base_offset - 1` (if not compacted)